mungegithub: Read alias files from repositories

apelisse opened the pull request:
Add command line flag so that we parse the files properly.

<!-- Reviewable:start -->

---
This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/contrib/2003)
<!-- Reviewable:end -->

Commit:
mungegithub: Read alias files from repositories

foxish commented on mungegithub/submit-queue/deployment/kubernetes/configmap.yaml:
```diff
@@ -44,3 +44,4 @@ data:
   check-labels.label-file: "/gitrepos/kubernetes/labels.yaml"
   issue-categorizer.triager-url: "http://issue-triager-service:5000"
   cla.cla-status-context: ""
+  alias.alias-file: "/gitrepos/kubernetes/OWNERS_ALIASES"
```
Are we going to add this file to all 3 repos before we merge this PR?

apelisse replied:
No. But I would expect the code to gracefully handle the missing file (rather than kubernetes gracefully handling the missing environment variable :-)

foxish replied:
We should probably change [mungegithub/features/aliases.go#L96](https://github.com/kubernetes/contrib/blob/b70447aa59ea14468f4cd349760e45b6a0a9b15d/mungegithub/features/aliases.go#L96).

foxish commented:
LGTM after we ensure that the alias feature handles the absence of a given file-path gracefully.

apelisse commented:
I'll take care of that, thanks @foxish

Commit:
mungegithub: Ignore missing ALIAS file

Ignore if the file is not present and can't be read. The mungebot
configuration should be able to say where the file is expected to be
even if it's not there yet.

Whenever the file fails to be read because it's not present, let's just
reset the current alias mapping to an empty mapping.

foxish commented on mungegithub/features/aliases.go:
```diff
@@ -92,6 +93,14 @@ func (a *Aliases) EachLoop() error {
 	// read and check the alias-file.
 	fileContents, err := a.aliasReader.read()
+	if os.IsNotExist(err) {
+		glog.Info("Missing alias-file (%s), using empty alias structure.", a.AliasFile)
```
Infof?

foxish commented:
You can self-apply LGTM after fixing nit. :)
Thanks!
